cirq/ops/controlled_test_util.py

```diff
@@ -0,0 +1,54 @@
+from typing import Union, Tuple, cast
+
+import numpy as np
+import cirq
+from cirq.type_workarounds import NotImplementedType
+
+
+class GateUsingWorkspaceForApplyUnitary(cirq.SingleQubitGate):
+    def _apply_unitary_(self, args: cirq.ApplyUnitaryArgs
+                        ) -> Union[np.ndarray, NotImplementedType]:
+        args.available_buffer[...] = args.target_tensor
+        args.target_tensor[...] = 0
+        return args.available_buffer
+
+    def _unitary_(self):
+        return np.eye(2)
```

Comment:
raise an exception instead of returning a bad result

Reply:
ack

Reply:
actually - I'm not sure what you mean exactly here. 1) this was a pre-existing dummy gate class implementation that I reused  2) it does seem to need to return some kind of valid matrix otherwise it fails to be consistent